src/components/Filter/Filter.js

```diff
@@ -0,0 +1,193 @@
+import "./Filter.css";
+import {useProduct} from "../../contexts/product-context"
+import { useState } from "react";
+
+
+export function Filter() {
+
+    const { state, dispatch } = useProduct();
+   
```

Comment:
Just a small suggestion, we could have multiple custom hooks hence naming state and dispatch would not be feasible try to use names like `productState, productDispatch` so that it can be more informative.

Reply:
ohh okay, thanks for pointing this out👍